cross-chain/arbitrum/deploy_l2/13_transfer_proxy_admin_ownership.ts

```diff
@@ -0,0 +1,25 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { helpers, upgrades, deployments } = hre
+  const { governance, deployer } = await helpers.signers.getNamedSigners()
+  const { log } = deployments
+
+  const proxyAdmin = await upgrades.admin.getInstance()
+  const currentOwner = await proxyAdmin.owner()
+
+  // The `@openzeppelin/hardhat-upgrades` plugin deploys a single ProxyAdmin
+  // per network. We don't want to transfer the ownership if the owner is already
+  // set to the desired address.
+  if (!helpers.address.equal(currentOwner, governance.address)) {
+    log(`transferring ownership of ProxyAdmin to ${governance.address}`)
+    await (
+      await proxyAdmin.connect(deployer).transferOwnership(governance.address)
+    ).wait()
+  }
+}
+
+export default func
+
+func.tags = ["TransferProxyAdminOwnership"]
```

Comment:
I think we may need force the order of this script by making it dependent on `ArbitrumTBTC` and `ArbitrumWormholeGateway` and using `runAtTheEnd = true`

Reply:
Not sure if I'm following. `ArbitrumTBTC` and `ArbitrumWormholeGateway` have 01_* and 02_* indexes respectively in the `deploy` dir, whereas this guy has 13_* index. Deployment scripts are executed in ascending order in our case, meaning that by the time we reach index 13, `ArbitrumTBTC` and `ArbitrumWormholeGateway` should be already deployed. Why do we need to add `runAtTheEnd`?

Reply:
This is true but relying on alphabetical order is dangerous. First, deployment scripts evolve so they can be modified, removed or new scripts can be added. Without dependencies stated in an explicit form (`runAtEnd` and `dependencies`) it's easy to break something during modifications. Moreover, during mainnet deployment, those scripts are typically run individually, so each script must define its dependencies to have the necessary prerequisites set before doing their work.

Reply:
I've named files to create a "buffer" for files that deploy stuff 1-10, transfer ownerships 11-20 and do authorization 21-30 so everything should run in order. But agree, it certainly won't hurt adding an extra guard by adding `runAtEnd` in those scripts that are meant to be run at the end. [6f718fe](https://github.com/keep-network/tbtc-v2/pull/554/commits/6f718feb665a13aa87b10e183cda2641cdf01eac)